semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/azureaisearch/AzureAISearchVectorStoreCollectionSearchMapping.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.connectors.data.azureaisearch;
+
+import com.microsoft.semantickernel.data.vectorsearch.options.BasicVectorSearchFilter;
+import com.microsoft.semantickernel.data.vectorsearch.options.EqualityFilterClause;
+import com.microsoft.semantickernel.data.vectorsearch.options.TagListContainsFilterClause;
+import com.microsoft.semantickernel.exceptions.SKException;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AzureAISearchVectorStoreCollectionSearchMapping {
+    public static String buildFilterString(BasicVectorSearchFilter basicVectorSearchFilter,
+        Map<String, String> storageNames) {
+        if (basicVectorSearchFilter == null) {
+            return "";
+        }
+
+        String filter = "";
+
+        if (basicVectorSearchFilter.getFilterClauses() != null) {
+
+            List<String> filterStrings = basicVectorSearchFilter.getFilterClauses().stream()
+                .filter(filterClause -> filterClause instanceof EqualityFilterClause)
+                .map(filterClause -> {
+                    String fieldName = ((EqualityFilterClause) filterClause).getFieldName();
```

Comment:
I think the correct api here, should be that FilterClause interface should have some method like:

```
 String formFilterQuery(String fieldName, String fieldValue);
```

Then EqualityFilterClause.formFilterQuery applies the logic here.

Reply:
I understand the point. However, FilterClause can be used for other connectors different to Azure AI Search.